### PR TITLE
Vendored scripts

### PR DESCRIPTION
**Pull request by jameshadfield**
A draft PR to show how vendored files could be consumed by pathogen repos

See https://github.com/nextstrain/ingest/pull/2 for the PR which add the script to the ingest repo, vendored here as `ingest/vendored`

See https://github.com/nextstrain/ingest/issues/3 for the `git subtree` commands I used

Note that I probably won't merge this PR, instead I'll recreate the subtree from its `main` branch once the above PR has been merged.

### CODE REVIEW

File: ingest/ingest.smk
Comment (victorlin) [inline ingest/ingest.smk:1]: Suggestions:

1. Add a section in the pathogen repo README on how to update these scripts with `git subtree pull`
2. Add that command in commit messages such as 7d507f7392f22144900ac3427106709b71ce3270 where those changes are the result of pulling new contents from the subtree remote.

File: ingest/vendored/apply-geolocation-rules [hunk 1]
```diff
@@ -5,18 +5,6 @@
 any additional transformations on top of the user curations.
 """
 
-"""
-Copied from https://github.com/nextstrain/monkeypox/blob/62fca491c6775573ad036eedf34b271b4952f2c2/ingest/bin/apply-geolocation-rules
-with two changes:
```
Comment (victorlin) [inline ingest/vendored/apply-geolocation-rules:10]: > The script vendored here contains two code changes

A comment is used to describe the pathogen-specific changes, when those might be better described by git history.

Suggestion: Put these changes in a separate commit. I'd expect something like one commit to add the original contents from `git subtree pull`, then another commit with the pathogen-specific modifications.

File: ingest/vendored/README.md
Comment (victorlin) [inline ingest/vendored/README.md:1]: I wish the auto-generated messages (f9d64076768e799619da8dfdff4581646b3db285, ee78e397f75d419c949b3f4e65e0080008e2badb) had more information on the remote. If I'm reading the git history alone, it's not clear where these come from.

Suggestion: Modify one of the auto-generated messages to include the remote URL, https://github.com/nextstrain/ingest. I'm not sure which is best since the remote commit hash is in f9d64076768e799619da8dfdff4581646b3db285, but only the merge commit ee78e397f75d419c949b3f4e65e0080008e2badb will show up as the one making changes.

**Review by victorlin — COMMENTED** [review]
Thanks for showing a clear example!

You said you wouldn't merge this PR, but I assume you can force-push to the same PR branch with a new subtree, or take these suggestions for a similar future PR.

File: ingest/vendored/apply-geolocation-rules (hunk 1 above)
Comment (victorlin) [reply ingest/vendored/apply-geolocation-rules:10]: Oh, I just realized that 7d507f7392f22144900ac3427106709b71ce3270 is a direct copy of https://github.com/nextstrain/ingest/commit/5afbcc788de7a54aafebcc0e2c701d81176a046a and these aren't pathogen-specific changes. You can disregard this suggestion then.

Comment (jameshadfield) [reply ingest/vendored/apply-geolocation-rules:10]: > I just realized that https://github.com/nextstrain/hepatitisB/commit/7d507f7392f22144900ac3427106709b71ce3270 is a direct copy of https://github.com/nextstrain/ingest/commit/5afbcc788de7a54aafebcc0e2c701d81176a046a

Yeah, or perhaps the other way around! The ingest commit was created by pushing the subtree from this (hbv) repo. In hindsight, probably not an approach I'd reach for again, but it was a good learning exercise.

**Comment by jameshadfield** [comment]
Superseded by #5
